lab.R

```diff
@@ -90,52 +98,94 @@ data("nytcovcounty")
 
 #' 1. *Write a pipe that starts with `nytcovcounty` as input, filters down to California and April 1-August 31, 2020, and assigns the result to a variable `filtered_df`.  Hint: You can compare dates as though they were strings, e.g., `date <= '1980-05-17'` gives you dates on or before May 17, 1980.* 
 #' 
+filtered_df = nytcovcounty %>%
+  filter(state == 'California')%>%
+  filter(date >= '2020-04-01', date<= '2020-08-31')
+filtered_df
 
 #' 2. To go from daily changes to cumulative counts, we'll use the following function. 
 
-daily_diff = function(x, order_var) {
-    diff = x - lag(x, order_by = order_var)
+daily_diff = function(x, date) {
+    diff = x - lag(x, order_by = date)
     return(diff)
 }
 
 #' *Write a pipe that takes `filter_df` as input, groups the data by county and FIPS code, sorts the dataframe by date, and then converts the cumulative cases and death counts to daily changes using `daily_diff`.  (`date` is the order variable.)  Assign the result to `daily_df`. Hint: `mutate()` can replace the value of existing variables.* 
 #' 
+daily_df = filtered_df %>%
+  group_by(county, fips) %>%
+  arrange(date) %>%
+  mutate(deaths = daily_diff(deaths, date),
+         cases = daily_diff(cases, date))
```

Comment:
Remember to `ungroup()` after grouping
```suggestion
daily_df = filtered_df %>%
  group_by(county, fips) %>%
  arrange(date) %>%
  mutate(deaths = daily_diff(deaths, date),
         cases = daily_diff(cases, date)) %>%
   ungroup()
```